bug: Non-Deterministic Test EntityIsolationIT.aclRuleWithResourceTypeIsolation

A test failed non-deterministically in GitHub Actions:

## failed test

`io.kroxylicious.it.filter.entityisolation.EntityIsolationIT.aclRuleWithResourceTypeIsolation(ResourcePattern, ResourcePattern, List, KafkaCluster)[2]`

https://github.com/kroxylicious/kroxylicious/actions/runs/24478633738/job/71538061248?pr=3724

## stack trace

```
[ERROR] Tests run: 19, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 66.87 s <<< FAILURE! -- in io.kroxylicious.it.filter.entityisolation.EntityIsolationIT
[ERROR] io.kroxylicious.it.filter.entityisolation.EntityIsolationIT.aclRuleWithResourceTypeIsolation(ResourcePattern, ResourcePattern, List, KafkaCluster)[2] -- Time elapsed: 1.273 s <<< FAILURE!
java.lang.AssertionError: 

Expecting actual:
  []
to contain exactly in any order:
  ["BobGroup"]
but could not find the following elements:
  ["BobGroup"]

	at io.kroxylicious.it.filter.entityisolation.EntityIsolationIT.aclRuleWithResourceTypeIsolation(EntityIsolationIT.java:709)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at java.base/java.util.Optional.ifPresent(Optional.java:178)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:184)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:184)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:184)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
	at java.base/java.util.Spliterators$ArraySpliterator.forEachRemaining(Spliterators.java:1024)
	at java.base/java.util.stream.ReferencePipeline$Head.forEach(ReferencePipeline.java:762)
	at java.base/java.util.stream.ReferencePipeline$7$1.accept(ReferencePipeline.java:276)
```

The test expected to find "BobGroup" in the results but received an empty list, suggesting a potential timing or synchronization issue in the entity isolation filter.

--- a/kroxylicious-integration-tests/src/test/java/io/kroxylicious/it/filter/entityisolation/EntityIsolationIT.java
+++ b/kroxylicious-integration-tests/src/test/java/io/kroxylicious/it/filter/entityisolation/EntityIsolationIT.java
@@ -462,10 +462,13 @@ class EntityIsolationIT {
                 assertThat(bobAdmin.incrementalAlterConfigs(Map.of(configResource, setConfig)).all())
                         .succeedsWithin(Duration.ofSeconds(5));
 
-                assertThat(bobAdmin.listConfigResources(Set.of(ConfigResource.Type.GROUP), new ListConfigResourcesOptions()).all())
-                        .succeedsWithin(Duration.ofSeconds(5))
-                        .asInstanceOf(InstanceOfAssertFactories.list(ConfigResource.class))
-                        .containsExactly(configResource);
+                await().atMost(Duration.ofSeconds(30))
+                        .untilAsserted(() -> {
+                            assertThat(bobAdmin.listConfigResources(Set.of(ConfigResource.Type.GROUP), new ListConfigResourcesOptions()).all())
+                                    .succeedsWithin(Duration.ofSeconds(5))
+                                    .asInstanceOf(InstanceOfAssertFactories.list(ConfigResource.class))
+                                    .containsExactly(configResource);
+                        });
             }
         }
     }
@@ -700,13 +703,16 @@ class EntityIsolationIT {
                 assertThat(bobAdmin.createAcls(List.of(aclBinding)).all())
                         .succeedsWithin(Duration.ofSeconds(5));
 
-                assertThat(bobAdmin.describeAcls(new AclBindingFilter(new ResourcePatternFilter(ResourceType.ANY, null, PatternType.ANY),
-                        AccessControlEntryFilter.ANY)).values())
-                        .succeedsWithin(Duration.ofSeconds(5))
-                        .asInstanceOf(collection(AclBinding.class))
-                        .extracting(AclBinding::pattern)
-                        .extracting(ResourcePattern::name)
-                        .containsExactlyInAnyOrderElementsOf(List.of(bobResourcePattern.name()));
+                await().atMost(Duration.ofSeconds(30))
+                        .untilAsserted(() -> {
+                            assertThat(bobAdmin.describeAcls(new AclBindingFilter(new ResourcePatternFilter(ResourceType.ANY, null, PatternType.ANY),
+                                    AccessControlEntryFilter.ANY)).values())
+                                    .succeedsWithin(Duration.ofSeconds(5))
+                                    .asInstanceOf(collection(AclBinding.class))
+                                    .extracting(AclBinding::pattern)
+                                    .extracting(ResourcePattern::name)
+                                    .containsExactlyInAnyOrderElementsOf(List.of(bobResourcePattern.name()));
+                        });
             }
         }
     }

